update_sc_name/update_scientific_name.py

```diff
@@ -1,31 +1,52 @@
+import logging
 import os.path
 import shutil
 from argparse import ArgumentParser
 
 from ebi_eva_common_pyutils import command_utils
+from ebi_eva_common_pyutils.logger import logging_config
 from ebi_eva_common_pyutils.metadata_utils import get_metadata_connection_handle
-from ebi_eva_common_pyutils.pg_utils import execute_query
-from ebi_eva_common_pyutils.taxonomy.taxonomy import get_scientific_name_from_eva, normalise_taxon_scientific_name
+from ebi_eva_common_pyutils.pg_utils import execute_query, get_all_results_for_query
+from ebi_eva_common_pyutils.taxonomy.taxonomy import normalise_taxon_scientific_name, get_scientific_name_from_ensembl
 
+logging_config.add_stdout_handler()
+logger = logging_config.get_logger(__name__)
 
-def update_path_reference_sequence(taxonomy, old_sc_name, ref_seq_dir, new_sc_name):
+def get_scientific_name_from_eva(taxonomy_id, private_config_xml_file, profile):
+    with get_metadata_connection_handle(profile, private_config_xml_file) as pg_conn:
+        query = f"select scientific_name from evapro.taxonomy where taxonomy_id={taxonomy_id}"
+        result = get_all_results_for_query(pg_conn, query)
+        return result[0][0]
+
+
+def update_path_reference_sequence(taxonomy, old_sc_name, new_sc_name, ref_seq_dir, ref_seq_dir_name, retain_old_dir):
     nmz_old_sc_name = normalise_taxon_scientific_name(old_sc_name)
-    old_species_ref_dir = os.path.join(ref_seq_dir, nmz_old_sc_name)
+    old_species_ref_dir = os.path.join(ref_seq_dir, ref_seq_dir_name) \
+        if ref_seq_dir_name else os.path.join(ref_seq_dir, nmz_old_sc_name)
     nmz_new_sc_name = normalise_taxon_scientific_name(new_sc_name)
     new_species_ref_dir = os.path.join(ref_seq_dir, nmz_new_sc_name)
 
+    logger.info(f"normalize old scientific name : {nmz_old_sc_name}, old species ref dir: {old_species_ref_dir}, "
+                   f"normalize new scientific name: {nmz_new_sc_name}, new species ref dir: {new_species_ref_dir}")
+
+    if old_species_ref_dir == new_species_ref_dir:
+        logger.warning(f"new species ref dir is same as old species ref dir. No need to update path reference")
+        return
+
     if not os.path.exists(old_species_ref_dir):
-        raise Exception(f'For taxonomy {taxonomy}, could not find any directory with name '
-                        f'"{nmz_old_sc_name}" in reference sequence directory {ref_seq_dir}')
+        raise Exception(f'For taxonomy {taxonomy}, could not find any directory with name "{old_species_ref_dir}"')
     else:
         if os.path.exists(new_species_ref_dir):
+            logger.warning(f"new species reference directory exists. syncing files between old and new dirs")
             sync_command = f"rsync -a --ignore-existing {old_species_ref_dir}/ {new_species_ref_dir}/"
```

Comment:
```suggestion
            sync_command = f"rsync -av --ignore-existing {old_species_ref_dir}/ {new_species_ref_dir}/"
```
Could be useful for debugging....

Reply:
updated